### PR TITLE
Decouple train and test datasets in classification template

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

File: pytorch_toolkit/ote/ote/metrics/classification/reid.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 import os
 import subprocess
+import logging
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/metrics/classification/reid.py:19]: keep imports sorted

File: pytorch_toolkit/ote/ote/metrics/classification/reid.py [hunk 2]
```diff
@@ -61,8 +62,8 @@ def get_topk_dict(value, k=1):
         subprocess.run(
             f'python {REID_TOOLS}/main.py'
             f' --config-file {config_path}'
-            f' test.evaluate True'
             f' {update_config}'
+            f' test.evaluate True'
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/metrics/classification/reid.py:66]: what is a reason of this change?

File: pytorch_toolkit/ote/ote/modules/evaluators/reid.py [hunk 3]
```diff
@@ -42,9 +44,9 @@ def _evaluate_internal(self, config_path, snapshot, out, update_config, metrics_
         if os.path.islink(snapshot):
             snapshot = os.path.join(os.path.dirname(snapshot), os.readlink(snapshot))
 
-        update_config['classification.data_dir'] = Path(update_config['data.root']).name
-        update_config['data.root'] = Path(update_config['data.root']).parent
-        update_config = ' '.join([f'{k} {v}' for k, v in update_config.items() if len(str(v)) and len(str(k))])
+        data_path_args = f'--custom-roots {update_config[self.parameter_test_dir]} {update_config[self.parameter_test_dir]} --root _ '
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/modules/evaluators/reid.py:47]: what does `--root _` mean?

File: pytorch_toolkit/ote/ote/modules/evaluators/reid.py [hunk 4]
```diff
@@ -42,9 +44,9 @@ def _evaluate_internal(self, config_path, snapshot, out, update_config, metrics_
...
+        del update_config[self.parameter_test_dir]
+        update_config = data_path_args + ' '.join([f'{k} {v}' for k, v in update_config.items() if len(str(v)) and len(str(k))])
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/modules/evaluators/reid.py:49]: there is no need to use `len` I guess

File: pytorch_toolkit/ote/ote/modules/trainers/reid.py [hunk 5]
```diff
@@ -41,10 +44,10 @@ def __call__(self, config, gpu_num, out, update_config, tensorboard_dir):
...
         logging.info(f'Commandline:\n{" ".join(sys.argv)}')
 
-        update_config['classification.data_dir'] = Path(update_config['data.root']).name
-        update_config['data.root'] = Path(update_config['data.root']).parent
-        update_config = ' '.join([f'{k} {v}' for k, v in update_config.items() if len(str(v)) and len(str(k))])
-
+        data_path_args = f'--custom-roots {update_config[self.parameter_train_dir]} {update_config[self.parameter_val_dir]} --root _ '
+        del update_config[self.parameter_train_dir]
+        del update_config[self.parameter_val_dir]
+        update_config = data_path_args + ' '.join([f'{k} {v}' for k, v in update_config.items() if len(str(v)) and len(str(k))])
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/ote/ote/modules/trainers/reid.py:50]: ditto

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

**Comment by Ilya-Krylov** [comment]
Not sure if export tests passed actually:

```
[2021-01-27T17:32:46.209Z] ONNX check passed.
[2021-01-27T17:32:46.209Z] OpenVINO Model Optimizer not found, please source openvino/bin/setupvars.sh before running this script.
```

File: pytorch_toolkit/ote/ote/modules/evaluators/reid.py (hunk 3 above)
Comment (sovrasov) [reply pytorch_toolkit/ote/ote/modules/evaluators/reid.py:47]: `root` actually is not required here, but it was added as a delimiter for argparse: if the last argument is a multiple-valued one (nargs=+), we can't fetch the remaining options: all of them are parsed as additional values. Now we have such arguments, so whey shouldn't be right before the remaining args

File: pytorch_toolkit/ote/ote/metrics/classification/reid.py (hunk 2 above)
Comment (sovrasov) [reply pytorch_toolkit/ote/ote/metrics/classification/reid.py:66]: Before this PR `update_config` has contained the remaining rags only, but now it has a regular args also, so they should go first.

File: pytorch_toolkit/ote/ote/metrics/classification/reid.py (hunk 1 above)
Comment (sovrasov) [reply pytorch_toolkit/ote/ote/metrics/classification/reid.py]: Fixed

File: pytorch_toolkit/ote/ote/modules/evaluators/reid.py (hunk 4 above)
Comment (sovrasov) [reply pytorch_toolkit/ote/ote/modules/evaluators/reid.py]: Didn't know. Fixed

File: pytorch_toolkit/ote/ote/modules/trainers/reid.py (hunk 5 above)
Comment (sovrasov) [reply pytorch_toolkit/ote/ote/modules/trainers/reid.py]: ditto

**Comment by sovrasov** [comment]
Docs builder has failed due to unrelated reason:
```
Error response from daemon: manifest for registry.gitlab.com/openvino-pushbot/icv-infra/md_link_checker_ubuntu16:latest not found: manifest unknown: manifest unknown
```
Tests are passed.

**Review by Ilya-Krylov — APPROVED** [review]

**Pull request merged**